Tools: Added how to use Gstreamer Debugger (gst-debugger)

leemgs opened the pull request:
Fixed issue 132#issuecomment-439788784.

This commit is to explain how to use a Gstreamer Debugger tool to
introspect a gst-pipeline remotely with a graphical client.

**Changes proposed in this PR:**
* Version 2:
1. Added GStreamer-based totem player as an example

* Version 1:
1. Added How to use Gstreamer debugger
2. Added How to patch ver 0.90.0 to build on Ubuntu 16.04

Signed-off-by: Geunsik Lim <geunsik.lim@samsung.com>

--

leemgs commented:
* After this PR: 
![image](https://user-images.githubusercontent.com/82404/48691464-01e4af00-ec16-11e8-97e3-9e70f987ddf8.png)

ohsewon commented on tools/debugging/README.md:
```diff
@@ -57,3 +57,48 @@ $ eog ${number_of_test_case}.png
+$ git clone https://github.com/GNOME/gst-debugger.git
+$ cd gst-debugger
+$ git checkout 0.90.0
+$ ./audogen.sh
```
typo 'audogen' -> 'autogen'

ohsewon replied:
for './autogen.sh', need 'libprotobuf-c' package.
but, `sudo apt install libprotobuf-c` doesn't work.

leemgs replied:
Typo is fixed.   If you need meed "checking for ****" statement while running autogen.sh script.  You have to install required package. 

* In my own Ubuntu: 
```bash
 dpkg -l | grep protobuf
ii  libmirprotobuf3:amd64                         0.26.3+16.04.20170605-0ubuntu1.1             amd64        Display server for Ubuntu - RPC definitions
ii  libprotobuf-c-dev                             1.2.1-1                                      amd64        Protocol Buffers C static library and headers (protobuf-c)
ii  libprotobuf-c1                                1.2.1-1                                      amd64        Protocol Buffers C shared library (protobuf-c)
ii  libprotobuf-dev:amd64                         2.6.1-1.3                                    amd64        protocol buffers C++ library (development files)
ii  libprotobuf-lite9v5:amd64                     2.6.1-1.3                                    amd64        protocol buffers C++ library (lite version)
ii  libprotobuf9v5:amd64                          2.6.1-1.3                                    amd64        protocol buffers C++ library
ii  protobuf-c-compiler                           1.2.1-1                                      amd64        Protocol Buffers C compiler (protobuf-c)
ii  protobuf-compiler                             2.6.1-1.3                                    amd64        compiler for protocol buffer definition files
ii  python-protobuf                               2.6.1-1.3                                    amd64        Python bindings for protocol buffers
```

leemgs replied:
Note: 
> GStreamer's plugin. This guide is written on Ubuntu 16.04 X86_64 distribution.

ohsewon replied:
What I needed is 'libprotobuf-c-dev'.

ohsewon replied:
Additionally need to install two packages, 'protobuf-compiler' and 'libboost-all-dev'

leemgs replied:
You describe  why packaging (e.g., .deb, .rpm) is important. :)

leemgs commented:
**Changes proposed in this PR:**
* Version 2:
1. Added GStreamer-based totem player as an example

* Version 1:
1. Added How to use Gstreamer debugger
2. Added How to patch ver 0.90.0 to build on Ubuntu 16.04
